DEVAI-38: Add Developer Hub Configuration Script (#10)
redhat-ai-dev/ai-rhdh-installer: README.md

This installer is incompatible with `default` namespace installations, install and uninstall commands must include `--namespace <target-namespace>` or the context namespace must be changed, e.g. `oc project <target-namespace>`.

## Configuration

To configure your RHDH/GitOps/Pipelines setup to use AI Software Templates, run the following:

> `NAMESPACE=<target-namespace> bash ./configure.sh`

Reviewer: Nit: If you don't set namespace and it auto defaults to `ai-rhdh` for example we should outline that here so that people don't need to set it if using default

Author: @Jdubrick Outlined this in https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10/commits/f79731d79bd8ba133e35bab0b0ec09d8dcc70445 changes.

This installer is incompatible with `default` namespace installations, install and uninstall commands must include `--namespace <target-namespace>` or the context namespace must be changed, e.g. `oc project <target-namespace>`.

## Configuration

> To configure your RHDH/GitOps/Pipelines setup to use AI Software Templates, run the following:

Reviewer: ```suggestion
For convenience you can configure your Red Hat Developer Hub, GitOps, and Pipelines Operators to enable the use of AI Software Templates by running the following:
```

`bash ./configure.sh`

> Please note if you changed the installation namespace used by the installer you will first need to run `export NAMESPACE=<namespace used>` as the default value is `ai-rhdh`.

Reviewer: ```suggestion
**Note**: If you changed the installation namespace used by the installer you will first need to run `export NAMESPACE=<namespace used>` as the default value is `ai-rhdh`.
```

> See the documents below on the configuration details for setting up each component.

Reviewer: ```suggestion
Alternatively, if you do not wish to use the all-in-one `configure.sh` script, you can find documentation below for configuring each component individually.
```

Detailed documentation for configuring Pipelines/Tekton can be found in [`PIPELINES-CONFIG.md`](./docs/PIPELINES-CONFIG.md)

> ## Developer Hub Configuration

Reviewer 2: I think we need to either

1. emphasize more on running these configuration scripts or,
2. streamline better through a main script like configure.sh

Otherwise, users will encounter errors when trying to login to RHDH via GitHub login.

I am trying to compare the UX between the rhtap-installer and this installer. rhtap-installer had two steps:

1. helm install/bash script calling helm install
2. running a PLR that was output from previous step 1 and it gives you callback url, webhook url, webhook secret, etc.

Author: > running a PLR that was output from previous step 1 and it gives you callback url, webhook url, webhook secret, etc.

@maysunfaisal  Done in https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10/commits/3a820df5e477b1d8c80c23fb8bdd2951e9527afe

Author: > streamline better through a main script like configure.sh

@maysunfaisal Provided in https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10/commits/4aec9db72f787370738cb19095c0e089e447906a

Author: @maysunfaisal Is there any other UX improvements that should be included in these changes?

Reviewer 2: it looks good now, thx

> **Note**: It is required to go through the steps under documentation mentioned in [GitOps/ArgoCD Configuration](#gitopsargocd-configuration) and [Pipelines/Tekton Configuration](#pipelinestekton-configuration) before the steps under the documentation below.

Reviewer: Should we have a main script that runs all 3 in sequence?

Author: Created the script in https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10/commits/4aec9db72f787370738cb19095c0e089e447906a

Reviewer 2: we should probably update the docs to say to run configure.sh then

Author: Created doc section in https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10/commits/c9c41273c392f968aebe98f7919a368ed6e2923d
